test/test-ctest.cpp

```diff
@@ -0,0 +1,126 @@
+#include "test-ctest.hpp"
+#include <boost/regex.hpp>
+#include <boost/make_shared.hpp>
+#include <iostream>
+
+using namespace icinga;
+
+BOOST_TEST_GLOBAL_FIXTURE(CTestFileGeneratorFixture);
+
+boost::unit_test::decorator::base_ptr CTestProperties::clone() const
+{
+	return boost::make_shared<CTestProperties>(m_Props);
+}
+
+CTestFileGenerator::CTestFileGenerator(const boost::filesystem::path& testexe, const boost::filesystem::path& outfile)
+	: m_Fp(outfile.string(), std::iostream::trunc), m_WorkDir(outfile.parent_path().string()),
+	  m_TestExe(testexe.string())
+{
+	m_Fp.exceptions(std::iostream::badbit | std::iostream::failbit);
+}
+
+void CTestFileGenerator::visit(boost::unit_test::test_case const& test)
```

Comment:
I know that the boost interface uses this syntax but I would use `const T&` here and all the overriden methods for consistency with our own code.

Reply:
I know this is really annoying but you've already adjusted most of them, though you've missed to adjust the `test_suite_start` and `test_suite_finish` methods in this file. Otherwise, I have nothing to complain and everything works fine!